core/resource/handler/v1beta1/resource.go

```diff
@@ -289,6 +290,43 @@ func (rh ResourceHandler) ChangeResourceNamespace(ctx context.Context, req *pb.C
 	return &pb.ChangeResourceNamespaceResponse{}, nil
 }
 
+func (rh ResourceHandler) ApplyResources(ctx context.Context, req *pb.ApplyResourcesRequest) (*pb.ApplyResourcesResponse, error) {
+	tnnt, err := tenant.NewTenant(req.GetProjectName(), req.GetNamespaceName())
+	if err != nil {
+		return nil, errors.GRPCErr(err, "invalid tenant details")
+	}
+
+	store, err := resource.FromStringToStore(req.GetDatastoreName())
+	if err != nil {
+		return nil, errors.GRPCErr(err, "invalid datastore Name")
+	}
+
+	if len(req.ResourceNames) == 0 {
```

Comment:
let's move this checking into the service layer

Reply:
Service layer actually checks for the actual number of resources. This is the request validation.